src/Components/Components/src/Routing/Router.cs

```diff
@@ -114,8 +125,33 @@ private static string StringUntilAny(string str, char[] chars)
                 : str.Substring(0, firstIndex);
         }
 
+        private void RefreshRouteTable()
+        {
+            var assemblies = AdditionalAssemblies == null ? new[] { AppAssembly } : new[] { AppAssembly }.Concat(AdditionalAssemblies);
+            var assembliesSet = new HashSet<Assembly>(assemblies);
+            if (_assemblies.Count != assembliesSet.Count)
+            {
+                Routes = RouteTableFactory.Create(assemblies);
+                _assemblies.UnionWith(assembliesSet);
+            }
+            // If the new assemblies set is the same length as the previous one,
+            // we check if they are equal-by-value and refresh the route table if
+            // not. NOTE: This is an uncommon occurrence and a cold-code path so
+            // we shouldn't end up needing to do an equal-by-value check too frequently.
+            if (_assemblies.Count == assembliesSet.Count)
```

Comment:
HashSet<T>.SetEquals will already make this check if the `other` is also a HashSet, so is this check needed?

Reply:
That sounds correct - looks like lines 132-141 (including the `if` check on 141) could be removed, leaving just lines 143-147.

Not a big deal as it won't affect either behavior or perf, but would be slightly fewer lines of code :)